src/psbt.cpp

```diff
@@ -138,6 +138,7 @@ CMutableTransaction PartiallySignedTransaction::GetUnsignedTx(bool force_unblind
         txout.scriptPubKey = *output.script;
 
         bool exp_value = output.m_value_commitment.IsNull() || force_unblinded;
+        exp_value = exp_value && output.amount != nullopt;
```

Comment:
Just an observation, not sure if there's anything to change, but: It feels like the CTxOutput interface is more intuitive here. (Where you have a field that is a three-way union of null/explicit/commitment.) And less prone to this kind of omission.

Reply:
I do not thing that a three-way tagged union implemneted as a byte vector is more intuitive than boolean logic on booleans.